bin/scaffold.js

```diff
@@ -0,0 +1,219 @@
+var Cli = require('nested-yargs');
+var Git = require('nodegit');
+var Wreck = require('wreck');
+var Bluebird = require('bluebird');
+var Path = require('path');
+var Rimraf = require('rimraf');
+var Colors = require('colors');
+var WriteFile = Bluebird.promisify(require('fs').writeFile);
+var JsdParse = require('comment-parser');
+var _ = require('lodash');
+var Create = require('./create');
+
+var ghWreck = Wreck.defaults({
+    baseUrl: 'https://api.github.com/',
+    headers: {
+        'User-Agent': 'wt-cli-scaffold'
+    },
+    json: true,
+});
+
+Bluebird.promisifyAll(ghWreck);
+
+module.exports = Cli.createCommand('scaffold', 'download webtask templates', {
+    params: '[webtask] [name]',
+    setup: function (yargs) {
+        Create.options.setup(yargs);
+    },
+    options: _.assign({}, Create.options.options, {
+        repo: {
+            description: 'git repo to use',
+            type: 'string',
+        },
+    }),
+    handler: handleScaffold,
+});
+
+var DEFAULT_REPO_NAME = 'auth0/wt-cli';
+var DEFAULT_REPO_SAMPLE_DIR = 'sample-webtasks/';
+
+function handleScaffold(argv) {
+    if(argv.params.webtask)
+        argv.params.webtask += /.js$/.test(argv.params.webtask) ? '' : '.js'; 
+    else
+        return listScaffolds()
+            .then(cleanup)
+            .catch(catchErr);
+
+    // If there's no slash or repo we assume they want to pull from the samples
+    if(!/\//.test(argv.params.webtask) && !argv.repo)
+        return getHead(DEFAULT_REPO_NAME, 'master')
+            .then(getTree)
+            .then(getBlob.bind(null, Path.join(DEFAULT_REPO_SAMPLE_DIR, argv.params.webtask)))
+            .then(scaffold);
+
+    var wtPath = argv.repo ?
+        argv.params.webtask :
+        argv.params.webtask
+            .split('/')
+            .slice(2)
+            .join('/');
+
+    if(argv.repo)
+        return getCommit(argv.repo, 'master')
```

Comment:
I didn't realize that it was designed for non-github git repos. Maybe this
could be piped through a tar gzip stream.

git archive --remote=git@github.com:foo/bar.git --prefix=path/to/ HEAD:path/to/


On Tue, Aug 18, 2015 at 6:15 PM Milo Mordaunt <notifications@github.com>
wrote:

> In bin/scaffold.js
> <https://github.com/auth0/wt-cli/pull/27#discussion_r37360565>:
>
> > +    // If there's no slash or repo we assume they want to pull from the samples
> > +    if(!/\//.test(argv.params.webtask) && !argv.repo)
> > +        return getHead(DEFAULT_REPO_NAME, 'master')
> > +            .then(getTree)
> > +            .then(getBlob.bind(null, Path.join(DEFAULT_REPO_SAMPLE_DIR, argv.params.webtask)))
> > +            .then(scaffold);
> > +
> > +    var wtPath = argv.repo ?
> > +        argv.params.webtask :
> > +        argv.params.webtask
> > +            .split('/')
> > +            .slice(2)
> > +            .join('/');
> > +
> > +    if(argv.repo)
> > +        return getCommit(argv.repo, 'master')
>
> I don't think so, since --repo is for non-github repos, so it can't use
> the API. I don't think I standard git URL exposes anything other than
> 'clone'.
>
> We could take out --repo entirely, because it's probably not necessary,
> but since the logic was there I left it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/auth0/wt-cli/pull/27/files#r37360565>.
>